libraries/redcore/api/hal/hal.php

```diff
@@ -882,6 +897,12 @@ public function apiTask()
 
 		$this->setData('result', $result);
 		$this->triggerFunction('displayErrors', $model);
+
+		if ($result === false)
```

Comment:
I dont think task should be 404. There exist tasks which you expect them to be true or false... this way if it is false, it will be 404 error...